lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -3168,6 +3178,16 @@ void VisitNestedScopes(ParseNode* pnodeScopeList, ParseNode* pnodeParent, ByteCo
                 // Push the param scope
                 byteCodeGenerator->PushScope(paramScope);
 
+                /*
+                We need this call to happen *before* visiting the param scope. If we have a split scope,
+                and this function has parameters that default to a lambda which uses the implicit "arguments" inside
+                of eval, we need this function's argumentsSymbol to be set already, so that we can mark it as
+                non-locally referenced inside of PostVisitFunction.
+
+                See Github issue #1209 for more.
+                */
+                AddVarsToScope(pnodeScope->sxFnc.pnodeVars, byteCodeGenerator);
```

Comment:
And indeed @aneeshdk and I discussed making the arguments symbol a normal symbol instead of a special case, to remove a lot of this extra work.  After some code review and discussion we concluded that there is no reason to have a specially handled arguments symbol and it should just work if we make it a normal symbol (with some special care to ensure it is available regardless whether the user script declared it or not).